runtime/src/main/java/org/corfudb/runtime/collections/CorfuStreamEntry.java

```diff
@@ -68,12 +75,26 @@ public CorfuStreamEntry(K key, V payload, M metadata, long address, OperationTyp
      * Convert a given SMREntry to CorfuStreamEntry.
      */
     public static <K extends Message, V extends Message, M extends Message>
-        CorfuStreamEntry<K, V, M> fromSMREntry(SMREntry entry, @Nonnull final Class<K> keyClass,
-            @Nonnull final Class<V> payloadClass, @Nullable final Class<M> metadataClass) {
+        CorfuStreamEntry<K, V, M> fromSMREntry(SMREntry entry, @Nonnull final long epoch,
+                                               @Nonnull final Class<K> keyClass,
+                                               @Nonnull final Class<V> payloadClass,
+                                               @Nullable final Class<M> metadataClass) {
 
         long address = entry.getGlobalAddress();
 
-        OperationType operationType = (entry.getSMRMethod().equals("put")) ? OperationType.UPDATE : OperationType.DELETE;
+        OperationType operationType;
+        log.trace("fromSRMEntry: Table {} streamer got SMR {}", keyClass.getName(), entry.getSMRMethod());
+        switch(entry.getSMRMethod()) {
+            case "put":
+                operationType = OperationType.UPDATE;
+                break;
+            case "clear":
+                operationType = OperationType.CLEAR;
+                break;
+            default:
```

Comment:
You might want to throw an exception on the default path, and have an explicit delete.

Reply:
good point, done!